### PR TITLE
feature: Twilio update

### PR DESCRIPTION
**Pull request by shamantraghav**
Closes  [DII-40](https://issues.corp.twilio.com/browse/DII-40)
Also contains a minor change for  [DI754](https://issues.corp.twilio.com/browse/DI-754)

Screenshot for:

Alert asking the user to update using the 'twilio update' command when a new update is available:
[DII-40](https://issues.corp.twilio.com/browse/DII-40)

image

Ocif doc: https://oclif.io/docs/releasing.html#autoupdater

Updating using the 'twilio update' command:

(The cli update fails as the installation was npm based. : https://github.com/oclif/plugin-update/issues/49 ) 
image

No autoupdate warning when the update fails: 
Screenshot 2021-08-03 at 5 34 23 PM

Autoupdate waring for when we install new plugins: 
Screenshot for: [DI754](https://issues.corp.twilio.com/browse/DI-754)
image

# Fixes #

A short description of what this PR does.

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [ ] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-cli/blob/main/CONTRIBUTING.md) and my PR follows them
- [ ] I have titled the PR appropriately
- [ ] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://twilio.com/help/contact), or create a GitHub Issue in this repository.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -111,6 +113,9 @@
       "plugins:preinstall": [
         "./src/hooks/plugin-install"
       ],
+      "prerun": [
```
Comment (shamantraghav) [inline package.json:116]: Adding a new prerun hook to warn users about the update feature not working for npm based installations: https://github.com/oclif/plugin-update/issues/49

File: src/hooks/plugin-install.js [hunk 2]
```diff
@@ -1,8 +1,15 @@
 const { logger } = require('@twilio/cli-core').services.logging;
+const chalk = require('chalk');
 
 const { isTwilioPlugin } = require('../services/plugins');
 
+const AUTOCOMLETE_INSTALL_WARNING = `If you’re using autocomplete, you’ll need to run ${chalk.bold(
```
Comment (shamantraghav) [inline src/hooks/plugin-install.js:6]: Autocomplete waring for the plugins:install command

File: src/hooks/prerun/install-prerun.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+/* eslint-disable no-console */
+const chalk = require('chalk');
+
+const UPDATE_ALERT = `Warning: Use '${chalk.bold('npm update twilio-cli')}' for npm based installations. `;
```
Comment (shamantraghav) [inline src/hooks/prerun/install-prerun.js:4]: Adding a new prerun hook to warn users about the update feature not working for npm based installations: oclif/plugin-update#49

**Review by kridai — APPROVED** [review]
👍

File: src/hooks/postrun/plugin-postrun.js [hunk 4]
```diff
@@ -1,16 +1,16 @@
...
-)} after installing a plugin and then open a new terminal window. The CLI needs to re-build its cache.`;
+)} after an update and then open a new terminal window. The CLI needs to re-build its cache.`;
 const AUTOCOMLETE_ALERT = `If you are running bash or zsh on macOS or Linux, you can run one of the two commands below (as appropriate for the shell you are using): \n '${chalk.bold(
   '1) twilio autocomplete bash',
 )}' or \n '${chalk.bold('2) twilio autocomplete zsh')}' `;
 
 module.exports = async function pluginPostRun(options) {
-  if (options.Command.id === 'plugins:update') {
-    console.warn(chalk.yellowBright(` » ${AUTOCOMLETE_WARNING}`));
+  if (options.Command.id === 'plugins:update' || options.Command.id === 'update') {
```
Comment (ravali-rimmalapudi) [inline src/hooks/postrun/plugin-postrun.js:12]: [Q] When `options.Command.id === 'update'` condition will be executed?

**Review by ravali-rimmalapudi — APPROVED** [review]
Looks Good.

File: src/hooks/postrun/plugin-postrun.js (hunk 4 above)
Comment (shamantraghav) [reply src/hooks/postrun/plugin-postrun.js:12]: This is the 'twilio update' command which I'm adding as part of this pull request. It updates the CLI for us. And once that is done, we need to warn the users about autocomplete as the cache needs to be rebuilt.

**Review by shrutiburman — APPROVED** [review]
Looks good! 🚀

**Pull request merged**